mirrord/cli/src/operator/session.rs

```diff
@@ -0,0 +1,150 @@
+                let not_supported = "`operator session kill-all` is not supported by the mirrord-operator found in your cluster, consider updating it!";
+
+                sub_progress.failure(Some(not_supported));
+                progress.failure(Some("Session operation `kill-all` failed!"));
+                error!(not_supported);
+            }
+            _ => (),
+        })
+        .map_err(|error| OperatorApiError::KubeError {
+            error,
+            operation: OperatorOperation::GettingStatus,
+        })
+        .map_err(CliError::from)?;
+
+    operator_session_finished(result.right(), sub_progress, progress);
+
+    Ok(())
+}
+
+/// `mirrord operator session kill --id {id}`: kills the operator session specified by `id`.
+#[tracing::instrument(level = "trace", ret)]
+pub(super) async fn operator_session_kill_one(id: u64) -> Result<()> {
+    let (mut progress, api, mut sub_progress) = operator_session_prepare().await?;
+
+    sub_progress.print("killing session with id {session_id}");
+
+    let result = api
+        .delete(&format!("{id}"), &DeleteParams::default())
+        .await
+        .inspect_err(|kube_fail| match kube_fail {
+            kube::Error::Api(response) if response.code == 404 && response.reason.contains("parse") => {
+                let not_supported = "`operator session kill` is not supported by the mirrord-operator found in your cluster, consider updating it!";
+
+                sub_progress.failure(Some(not_supported));
+                progress.failure(Some("Session operation `kill` failed!"));
+                error!(not_supported);
+            }
+            _ => (),
+        })
+        .map_err(|error| OperatorApiError::KubeError {
+            error,
+            operation: OperatorOperation::GettingStatus,
+        })
+        .map_err(CliError::from)?;
+
+    operator_session_finished(result.right(), sub_progress, progress);
+
+    Ok(())
+}
+
+/// `mirrord operator session kill {id}`: performs a clean-up for operator sessions that are still
+/// stored;
+#[tracing::instrument(level = "trace", ret)]
+pub(super) async fn operator_session_retain_active() -> Result<()> {
+    let (mut progress, api, mut sub_progress) = operator_session_prepare().await?;
+
+    sub_progress.print("retaining only active sessions");
+
+    let result = api
+        .delete("inactive", &DeleteParams::default())
+        .await
+        .inspect_err(|kube_fail| match kube_fail {
+            kube::Error::Api(response) if response.code == 404 && response.reason.contains("parse") => {
+                let not_supported = "`operator session retain-active` is not supported by the mirrord-operator found in your cluster, consider updating it!";
+
+                sub_progress.failure(Some(not_supported));
+                progress.failure(Some("Session operation `retain-active` failed!"));
+                error!(not_supported);
```

Comment:
This is always printed and looks kinda ugly, I'd remove it

Reply:
I'll use this box to address your review comment.

```sh
executing `mirrord operator kill-all`
  x Session management operation failed!
    x mirrord operator foobar does not support feature session management                                                                                                                                           Error:   × Feature `session management` is not supported in mirrord operator foobar.
  help: 
        
        - If you're still stuck:
        
        >> Please open a new bug report at https://github.com/metalbear-co/mirrord/issues/new/choose
        
        >> Or join our Discord https://discord.gg/metalbear and request help in #mirrord-help
        
        >> Or email us at hi@metalbear.co
```

Doesn't this error look beautiful now or what? It's just missing the proper operator version (wip).

Reply:
> 1. Do we really need the response.reason.contains("parse") part when checking operator response? I think checking status code should be enough

I think we do need it. I'm filling the error in the operator side, so now we're going to have nicer error messages coming from there (at least for this feature), and as such, we need to distinguish from the "random generic kube api error due to version mismatch", and the proper `reason` errors being returned.

Reply:
```sh
executing `mirrord operator kill-all`
  x Session management operation failed!
    x mirrord operator 3.76.0 does not support feature session management                                                                                                                                           Error:   × Feature `session management` is not supported in mirrord operator 3.76.0.
  help: 
        
        - If you're still stuck:
        
        >> Please open a new bug report at https://github.com/metalbear-co/mirrord/issues/new/choose
        
        >> Or join our Discord https://discord.gg/metalbear and request help in #mirrord-help
        
        >> Or email us at hi@metalbear.co

```

We have version!

Reply:
And the error in case we don't find an id to remove:

```
executing `mirrord operator kill --id 257`
  x Session operation failed!
    x `mirrord operator kill --id 257` failed due to `Session 257 not found!` with code `404`!                                                                                                                      Error:   × Operator returned a failure status for `mirrord operator kill --id 257`!
  help: Session 257 not found!
```